pkgdb/src/buildenv/realise.cc

```diff
@@ -48,7 +48,7 @@ namespace flox::buildenv {
 static const std::string BASH_ACTIVATE_SCRIPT = R"(
 # We use --rcfile to activate using bash which skips sourcing ~/.bashrc,
 # so source that here.
-if [ -f ~/.bashrc ]
+if [ -f ~/.bashrc -a "${FLOX_SOURCED_FROM_SHELL_RC:-}" != 1 ]
```

Comment:
question blocking: do we need something similar for zsh?

Reply:
afaict for zsh the shell init and activation is already separated.
shell init happens through the files in `flox.zdotdir` which call the activation script.
in the bash case everything happens in the bashrc file ( `--bashrc $FLOXENV/activate/bash`).
I wonder why we couldn't just call the common init script from the custom bashrc though like we do in zsh..